openweathermap-ios/openweathermap-ios/BookmarkLocation/Manager/BookmarkLocationManager.swift

```diff
@@ -0,0 +1,54 @@
+import Foundation
+import os.log
+
+class BookmarkLocationManager {
+
+    func save(bookmarkLocation: BookmarkLocation) throws -> [BookmarkLocation] {
+        var allBookmarks = try getAll()
+        let defaults = UserDefaults.standard
+        allBookmarks.append(bookmarkLocation)
+        let archiveData = try self.archiveData(allBookmarks)
+        defaults.set(archiveData, forKey: Bookmark.location.rawValue)
+        return allBookmarks
+    }
+
+    func getAll() throws -> [BookmarkLocation] {
+        let bookmarks: [BookmarkLocation] = (try? unArchiveData()) ?? .init()
+        return bookmarks
```

Comment:
```suggestion
        try? unArchiveData() ?? .init()
```

Reply:
In this case I'm needing to wrapper `try? unArchiveData()` with `()` or I'm getting this error
<img width="1444" alt="image" src="https://user-images.githubusercontent.com/5117106/80482588-d7c70e80-892a-11ea-9a21-3eb28de725e4.png">

Now that's ok

<img width="1460" alt="image" src="https://user-images.githubusercontent.com/5117106/80482701-10ff7e80-892b-11ea-85ea-b1d55419d9d9.png">

Reply:
Fixed 212f90e but still with wrapper `()`